modules/material-management/test/cryptographic_material.test.ts

```diff
@@ -163,7 +180,14 @@ describe('decorateCryptographicMaterial', () => {
     material.setUnencryptedDataKey(dataKey, { keyNamespace: 'k', keyName: 'k', flags: KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY })
     const test = material.getUnencryptedDataKey()
     test[0] = 12
-    expect(() => material.getUnencryptedDataKey()).to.throw()
```

Comment:
If I understand the code below, we are testing that one way to try to modify a KeyObject udk doesn't work. The purpose of this test should be to test that if it does get modified, then we fail. If it should be impossible to modify the KeyObject, then perhaps the below is fine for the KeyObject case, however I still think we should check that is the udk is not a KeyObject then it still correctly fails if modified.

Reply:
In the non-KeyObject case `getUnencryptedDataKey` will throw before the `dataKey` is even returned.  In this case we are dealing with Uint8Array (raw memory).  The cryptographic material keeps an independent copy of the `dataKey` and compares it to the `dataKey` that it is going to return.

This is because _all_ returned instances are references to the same memory so anyone _could_ have manipulated the memory.